cmd/contour/serve.go

```diff
@@ -235,13 +235,11 @@ func NewServer(log logrus.FieldLogger, ctx *serveContext) (*Server, error) {
 					// Other types of Secrets will never be referred to, so we can remove all data.
 					// For example Secrets of type helm.sh/release.v1 can be quite large.
 					// Last-applied-configuration annotation might contain a copy of the complete data.
-					secret.Data = nil
+					secret.Data = map[string][]byte{}
 					secret.SetManagedFields(nil)
 					secret.SetAnnotations(nil)
 
-					// Returning error will avoid handlers from being called.
-					// This does not avoid caching.
-					return nil, fmt.Errorf("ignoring secret %s/%s of type %s", secret.Namespace, secret.Name, secret.Type)
+					return secret, nil
```

Comment:
we can't return an error here with the updated handling of the transformer func, see: https://github.com/kubernetes/client-go/commit/fffc68d58e65d023e9783824b1c0fe1c2468c11e

Reply:
Looks consistent with what we did when we upgraded to 1.27 deps (https://github.com/projectcontour/contour/pull/5362), 👍

Reply:
contour informers don't sync and this error is repeatedly logged without this

`time="2023-07-26T20:27:34Z" level=error msg="pkg/mod/k8s.io/client-go@v0.26.7/tools/cache/reflector.go:169: Failed to watch *v1.Secret: unable to sync list result: couldn't enqueue object: ignoring secret kube-system/bootstrap-token-abcdef of type bootstrap.kubernetes.io/token\n" caller="reflector.go:140" context=kubernetes error="<nil>"`